include/xtypes/Module.hpp

```diff
@@ -385,6 +393,8 @@ class Module
     }
 
 protected:
+    friend std::string idl::generator::module_contents(const Module& module, size_t tabs);
```

Comment:
Why is necessary this friend? As I can see, you have all the types of the module publicly accessible.

Reply:
If you take care, the `generator::module_contents` access to the maps directly to avoid an ugly and potentially huge switch. It avoids copies too.

Reply:
ok...